src/tools/clippy/clippy_lints/src/precedence.rs

```diff
@@ -1,38 +1,17 @@
 use clippy_utils::diagnostics::span_lint_and_sugg;
 use clippy_utils::source::snippet_with_applicability;
-use rustc_ast::ast::{BinOpKind, Expr, ExprKind, MethodCall, UnOp};
-use rustc_ast::token;
+use rustc_ast::ast::{BinOpKind, Expr, ExprKind};
 use rustc_errors::Applicability;
 use rustc_lint::{EarlyContext, EarlyLintPass};
 use rustc_session::declare_lint_pass;
 use rustc_span::source_map::Spanned;
 
-const ALLOWED_ODD_FUNCTIONS: [&str; 14] = [
```

Comment:
Are we ok w/ getting rid of this list? It seemed a bit weird, but does seem to suppress some of the cases where it's mathematically irrelevant i.e. `-sin(x) = sin(-x)`

Reply:
ralf asked for the lint to be consistent in https://github.com/rust-lang/rust/pull/117161#discussion_r1371859160

Reply:
ok thanks! im ok w that too